types/glint.d.ts

```diff
@@ -43,6 +43,54 @@ declare module '@glint/environment-ember-loose/registry' {
     'sortable-handle': ModifierLike<{ Args: { Positional: [] } }>;
     sub: HelperLike<{ Args: { Positional: [number, number] }; Return: number }>;
     'svg-jar': ComponentLike<{ Args: { Named: { class: string }; Positional: [string] } }>;
+    CodeMirror: ComponentLike<{
+      Element: HTMLElement;
+      Args: {
+        Named: {
+          // Document options
+          document?: string;
+          onUpdate?: (newValue: string) => void;
```

Comment:
```suggestion
          onDocumentUpdate?: (newDocument: string) => void;
```

(Makes it more clear what exactly changed)

Reply:
@VasylMarchuk when you convert this to a `.ts` component, could you a short (one-line will do) documentation for each option? 

I found your notes in the PR description super useful: 

<img width="861" alt="Screenshot 2024-05-07 at 20 12 17" src="https://github.com/codecrafters-io/frontend/assets/3893573/f14fc330-6402-4825-a2a3-cef996bb311b">

Just copying these in verbatim should do! Else we'll have to keep coming back this to PR or the codemirror docs to figure out what options do.

Reply:
Renamed `onUpdate` to `documentUpdate` already, adding JSDocs now